pkg/odo/cli/deploy/deploy.go

```diff
@@ -41,6 +44,28 @@ func (o *DeployOptions) Complete(name string, cmdline cmdline.Cmdline, args []st
 	if err != nil {
 		return err
 	}
+
```

Comment:
Perhaps you could add a cleanup defer in case this method errors out. Also, consider adding the cleanup in `Run` method.

Reply:
@feloy?